Python/compile.c

```diff
@@ -1047,6 +1051,24 @@ compiler_addop_name(struct compiler_unit *u, location loc,
         arg <<= 1;
         arg |= 1;
     }
+    if (opcode == LOAD_SUPER_ATTR) {
+        arg <<= 2;
+    }
+    if (opcode == LOAD_SUPER_METHOD) {
+        opcode = LOAD_SUPER_ATTR;
+        arg <<= 2;
+        arg |= 1;
+    }
+    if (opcode == LOAD_ZERO_SUPER_ATTR) {
+        opcode = LOAD_SUPER_ATTR;
+        arg <<= 2;
+        arg |= 2;
+    }
+    if (opcode == LOAD_ZERO_SUPER_METHOD) {
+        opcode = LOAD_SUPER_ATTR;
+        arg <<= 2;
```

Comment:
Pseudo ops are currently used for things that get resolved to real ops only after optimisations.

Reply:
I chose this approach in order to parallel the existing use of `LOAD_METHOD` pseudo-op, resolved here to `LOAD_ATTR` with an oparg bit flag. The parallel between `LOAD_ATTR` and `LOAD_SUPER_ATTR` is so close that it seems to me that they should use the same technique. Would you prefer if this PR adds a helper function for `compiler_addop_name_bitflag` and uses that for all these cases, removing also the `LOAD_METHOD` pseudo-op?

Reply:
I see. I think we should do what you suggest, either in this PR or in a followup one.

Reply:
I'll submit it as a follow-up PR, since it's changing the existing `LOAD_ATTR` pattern to a new one.

Reply:
@iritkatriel I started working on this, but I'm not convinced it's going to be an improvement. The problem is that we can't do what you suggested:

> write a function that takes a local enum value, calculates the arg, and calls this function with the real opcode.

because the whole purpose of "this function" (`compiler_addop_name`) is to calculate the oparg value by adding a name to a names dict; we don't call this function with an oparg value that we could pre-shift. So to do this with a helper function (and not pseudo-ops), the helper function would have to be a copy-paste-modify duplicate of `compiler_addop_name` that is used only for `LOAD_ATTR`. Or else `compiler_addop_name` would have to directly take an extra enum arg (set to an unused default whenever non `LOAD_ATTR`) to represent the desired bit shift and bit flags to set. Both of these seem inferior to me to the existing pseudo-op approach, but I may not be seeing all the costs of the current approach? If you think one of these is still preferable (or there's a better option I'm not seeing), let me know and I'll put up a PR.

Reply:
cc @Fidget-Spinner, who did the consolidation of `LOAD_METHOD` into `LOAD_ATTR`, and may have additional context on why a pseudo-op was chosen at that time.